mods/default/functions.lua

```diff
@@ -715,6 +715,48 @@ function default.register_craft_metadata_copy(ingredient, result)
 	end)
 end
 
+--
+-- Log API / helpers
+--
+
+local log_non_player_actions = minetest.settings:get_bool("log_non_player_actions", false)
+
+-- `vector.check` is available since minetest >= 5.5.0
+local is_vector = vector.check or function(v)
```

Comment:
You can directly use `vector.check` everywhere, backwards compatibility is not necessary.

Reply:
okay, then i have to update my (test-) minetest server from 5.4.1 to 5.5.0 first ...

Reply:
In many places in minetest_game we still work with pos as a table (as described in https://github.com/minetest/minetest_game/blob/edd033b708488472c6bd688b14d8a5f59ef5b218/game_api.txt#L16). 

You can see it with `grep -rF '{x ' .`.

The use of `vector.check` breaks the previous compatibility to minetest 5.4.1.  In the README.md it should be pointed out that at least minetest 5.5.0 is needed.

Under these circumstances, I would find it more consistent and cleaner to do without `vector.check` and rename the function `is_vector(v)` to `is_pos(pos)`.

I think a general change from "pos" to "vector" should be done separately.

However, if we stick to `vector.check`, the use of `vector.to_string` instead of `minetest.pos_to_string` would be obvious. However, the messages would change here (space after comma):
`minetest.pos_to_string` returns `(X,Y,Z)`
`vector.to_string` returns `(X,Y,Z)`

Do you still insist on `vector.check` and what do you think about `vector.to_string`?

Reply:
> In many places in minetest_game we still work with pos as a table

Does `vector.check` not work on them?

> The use of vector.check breaks the previous compatibility to minetest 5.4.1. In the README.md it should be pointed out that at least minetest 5.5.0 is needed.

The readme is already correct on this. Commits merged into MTG *git* do not need to be compatible with MT *stable*.

> what do you think about vector.to_string?

I think it's better to keep the logging format without spaces.

Reply:
>>  In many places in minetest_game we still work with pos as a table

> Does `vector.check` not work on them?

An error would only occur if you pass pos as table like:
```lua
default.log_player_action(player, "does something", {x = 1, y = 2, z = 3})
```
But we don't have such a constellation in minetest_game at the moment.

I just find it, as mentioned above, a bit inconsistent that the function expects a vector instead of pos as a table.
But i have now adjusted it as desired: https://github.com/minetest/minetest_game/pull/2941/commits/1ba876893dc9f9c37e8c436cddda4c56bce1a1ab ;-)

Reply:
I am very happy with this https://github.com/minetest/minetest_game/pull/2941/commits/a6d940abda11e09342685893ec28b1acce20cf45 :sweat_smile: